### PR TITLE
Sync to Prod, Update Libraries, Update docker development

### PR DESCRIPTION
**Pull request by skyspook**
Large-ish PR, but codebase was in need up severe updates.
This accomplishes:

- Sync of code from production to github
- Use of OpenLDAP DMS AD repo to simulate AD
- Update to a custom AD integration and away from abandoned software
- Update of various libraries to mitigate security concerns
- Update of Docker environment to supported images and setup for full development and debugging

### CODE REVIEW

File: config/app.testing.php [hunk 1]
```diff
@@ -398,56 +380,13 @@
...
      */
     'ActiveDirectory' => [
-        'account_suffix' => '@dev.com',
-        'admin_username' => 'AD_ADMIN',
-        'admin_password' => 'AD_ADMIN_PW',
-        'base_dn' => 'DC=dev,DC=com',
-        'domain_controllers' => ['dc1.dev.com']
+        'username_field' => 'sAMAccountName',
+        'admin_username' => 'cn=admin,dc=dms,dc=local',
+        'admin_password' => 'Adm1n!',
```
Comment (denzuko) [inline config/app.testing.php:385]: Any way to prevent secret leaks like this?

File: docker-compose.yml [hunk 2]
```diff
@@ -1,43 +1,84 @@
...
+      image: mailhog/mailhog
+      restart: always
+      ports:
+        - 8025:8025
     db:
-        image: mariadb:5.5
+        image: mariadb:latest
         ports:
-            - "3306:3306"
+            - "53306:3306"
```
Comment (denzuko) [inline docker-compose.yml:28]: Can this port mapping just be removed and use network links?

File: docker-compose.yml [hunk 3]
```diff
@@ -1,43 +1,84 @@
...
+        entrypoint: ['/var/www/.docker/wait-for-it.sh', "db:3306", "-s", "-t", "120", "--" ]
+        command: '/var/www/.docker/startup.sh'
         environment:
           DEBUG: 'true'
           DB_HOST: 'db' # Leave this as 'db' to utilize MySQL container(s)
-          DB_USERNAME: 'root'
-          DB_PASSWORD: 'cakephp'
-          DB_DATABASE: 'cakephp'
+          DB_USERNAME: 'calendar'
+          DB_PASSWORD: 'calendar'
```
Comment (denzuko) [inline docker-compose.yml:74]: secret leak??

File: .gitignore [hunk 4]
```diff
@@ -10,3 +10,6 @@
 /.vscode
```
Comment (denzuko) [inline .gitignore:10]: syntax error?

File: Dockerfile [hunk 5]
```diff
@@ -1,41 +1,25 @@
-FROM hhvm/hhvm-proxygen:latest
-
-LABEL maintainer="infrastructure@dallasmakerspace.org"
-
-LABEL affinity:org.dallasmakerspace.application=infrastructure
-LABEL affinity:org.dallasmakerspace.architecture!=armhf
-
-LABEL org.dallasmakerspace.orgunit "Infrastructure Committee"
```
Comment (denzuko) [inline Dockerfile:8]: Labels are here for accounting, affinity, and CMDB tracking. Could we add them back?

File: Dockerfile [hunk 6]
```diff
@@ -1,41 +1,25 @@
...
-LABEL org.dallasmakerspace.environment "production"
-LABEL org.dallasmakerspace.application "infrastructure"
-LABEL org.dallasmakerspace.role "web application"
-LABEL org.dallasmakerspace.owner "infrastructure@dallasmakerspace.org"
-LABEL org.dallasmakerspace.customer "COMMITTEES:Infrastructure"
-LABEL org.dallasmakerspace.costcenter "FUNDS_MONTHLY:Web Hosting"
-LABEL org.dallasmakerspace.oid "iso.org.dod.internet.50391"
-LABEL org.dallasmakerspace.duns "iso.org.duns.053332191"
-
-ENV DEBIAN_FRONTEND noninteractive
```
Comment (denzuko) [inline Dockerfile:23]: Needed for best practice with debian images unless the base image already defines this. Comments?

File: Dockerfile [hunk 7]
```diff
@@ -1,41 +1,25 @@
...
+    a2enmod expires && \
+    a2enmod headers && \
+    a2enmod http2 && \
+    sed -e '/<Directory \/var\/www\/>/,/<\/Directory>/s/AllowOverride None/AllowOverride All/' -i /etc/apache2/apache2.conf
 
-# Reconfigure HHVM
-#ADD hhvm.prod.ini /etc/hhvm/site.ini
 
-EXPOSE 80
-HEALTHCHECK --interval=5m CMD 'curl -sSlk http://localhost/ || exit -1'
```
Comment (denzuko) [inline Dockerfile:41]: Can we add the healthcheck back in?

File: Dockerfile [hunk 8]
```diff
@@ -1,41 +1,25 @@
...
-LABEL org.dallasmakerspace.duns "iso.org.duns.053332191"
-
-ENV DEBIAN_FRONTEND noninteractive
+FROM php:7.4.20-apache-buster
 
 # Install Dependancies
-RUN apt-get update -y && apt-get install -y curl
+ADD https://github.com/mlocati/docker-php-extension-installer/releases/latest/download/install-php-extensions /usr/local/bin/
+RUN apt update -y && \
+    apt upgrade -y && \
```
Comment (denzuko) [inline Dockerfile:6]: Upgrades on a non base image is bad, adds layers and should be avoided.

File: Dockerfile [hunk 9]
```diff
@@ -1,41 +1,25 @@
...
+    apt install -y curl zip unzip git mariadb-client && \
```
Comment (denzuko) [inline Dockerfile:7]: should tag release versions. e.g. `curl==7.64.0-4 ...` Not a show stopper but a stretch goal.

File: docker-compose.yml [hunk 10]
```diff
@@ -1,43 +1,84 @@
...
-
 services:
-    ipam:
-        image: osixia/openldap:1.2.0
-        command: ["--copy-service", "--loglevel", "debug"]
-        volumes: ["/srv/ldap/schema:/etc/ldap/schema:rw"]
-
+    ldap:
+        extends:
+            file: ./dms-ad-openldap/docker-compose.yml
```
Comment (denzuko) [inline docker-compose.yml:4]: Just wanted to comment; nice use of submodules and docker-compose files. Side note, should ensure compose files are swarm safe.

File: docker-compose.yml [hunk 11]
```diff
@@ -1,43 +1,84 @@
...
+        links:
+        - ldap
+    mail:
+      image: mailhog/mailhog
+      restart: always
+      ports:
+        - 8025:8025
     db:
-        image: mariadb:5.5
+        image: mariadb:latest
```
Comment (denzuko) [inline docker-compose.yml:26]: unless shepard is use, one should define versions or better sha tags.

File: docker-compose.yml [hunk 12]
```diff
@@ -1,43 +1,84 @@
...
         ports:
-            - "3306:3306"
+            - "53306:3306"
         environment:
           MYSQL_ROOT_PASSWORD: 'cakephp'
-          MYSQL_DATABASE: 'calendarapp'
-
+          MYSQL_DATABASE: 'dms-calendar'
+          MYSQL_USER: 'calendar'
+          MYSQL_PASSWORD: 'calendar'
```
Comment (denzuko) [inline docker-compose.yml:33]: Secret leak?

File: docker-compose.yml [hunk 13]
```diff
@@ -1,43 +1,84 @@
...
+        networks:
+            main:
+                aliases:
+                    - db
+    phpmyadmin:
+        image: phpmyadmin/phpmyadmin
+        environment:
+            - PMA_ARBITRARY=1
+            - PMA_HOST=db
+            - PMA_USER=root
```
Comment (denzuko) [inline docker-compose.yml:43]: secret leak?!

**Review by denzuko — CHANGES_REQUESTED** [review]
Please review comments.

File: config/app.testing.php (hunk 1 above)
Comment (skyspook) [reply config/app.testing.php:385]: Not a secret leak, these are just the passwords for the docker DEV environment

File: docker-compose.yml (hunk 10 above)
Comment (skyspook) [reply docker-compose.yml:4]: Should really be named docker-compose.dev as it's only made for development environment. We're not running dockerized applications at this time and won't be working on that unless we have people to manage.

File: docker-compose.yml (hunk 11 above)
Comment (skyspook) [reply docker-compose.yml:26]: again, for dev only, and AWS keeps our mariadb up to latest builds, so it's appropriate.

File: docker-compose.yml (hunk 2 above)
Comment (skyspook) [reply docker-compose.yml:28]: Mostly kept here in case a developer wishes to use sql workbench instead of phpmyadmin.

File: Dockerfile (hunk 5 above)
Comment (skyspook) [reply Dockerfile:8]: Nope - these images are not maintained and we are not pushing them to a hub, so there's no reason to keep the labels. Our production envrionment is not containerized and unless we have people to support it, it's not going to happen right now.

**Comment by skyspook** [comment]
@denzuko  - while I appreciate the review, we're not going to be using docker for production anytime soon, and this codebase has essentially been abandoned. I'll be bringing it up to what's in production and this hopefully gets it to a point where it's approachable by people offering to help.

**Pull request closed without merging**